lndclient/lnd_services.go

```diff
@@ -192,29 +220,74 @@ func (s *GrpcLndServices) Close() {
 }
 
 // checkLndCompatibility makes sure the connected lnd instance is running on the
-// correct network.
+// correct network, has the version RPC implemented, is the correct minimal
+// version and supports all required build tags/subservers.
 func checkLndCompatibility(conn *grpc.ClientConn, chainParams *chaincfg.Params,
-	readonlyMac serializedMacaroon, network string) error {
+	readonlyMac serializedMacaroon, network string,
+	minVersion *verrpc.Version) (string, [33]byte, *verrpc.Version, error) {
 
-	lightningClient := newLightningClient(
-		conn, chainParams, readonlyMac,
-	)
+	// onErr is a closure that simplifies returning multiple values in the
+	// error case.
+	onErr := func(err error) (string, [33]byte, *verrpc.Version, error) {
```

Comment:
Instead of this, maybe `checkLndCompatibiltiy` (which now does more than just checking) can be an `init` method on `LndServices`, so that the retrieved data can be written into fields directly

Reply:
The `LndServices` is not instantiated here yet. I feel like I've refactored enough with this PR already.

Reply:
Fair enough